test/es-module/test-esm-type-field-errors.js

```diff
@@ -13,25 +14,45 @@ const packageTypeModuleMain =
   require.resolve('../fixtures/es-modules/package-type-module/index.js');
 
 // Check that running `node` without options works
-expect('', mjsFile, '.mjs file');
-expect('', cjsFile, '.cjs file');
-expect('', packageTypeModuleMain, 'package-type-module');
-expect('', packageTypeCommonJsMain, 'package-type-commonjs');
-expect('', packageWithoutTypeMain, 'package-without-type');
-
-// Check that --input-type isn't allowed for files
-expect('--input-type=module', packageTypeModuleMain,
-       'ERR_INPUT_TYPE_NOT_ALLOWED', true);
-
-try {
-  require('../fixtures/es-modules/package-type-module/index.js');
-  assert.fail('Expected CJS to fail loading from type: module package.');
-} catch (e) {
-  assert.strictEqual(e.name, 'Error');
-  assert.strictEqual(e.code, 'ERR_REQUIRE_ESM');
-  assert(e.toString().match(/require\(\) of ES Module/g));
-  assert(e.message.match(/require\(\) of ES Module/g));
-}
+describe('ESM type field errors', { concurrency: true }, () => {
+  it('.cjs file', () => {
+    expect('', cjsFile, '.cjs file');
+  });
+
+  it('.mjs file', () => {
+    expect('', mjsFile, '.mjs file');
+  });
+
+  it('package.json with "type": "module"', () => {
+    expect('', packageTypeModuleMain, 'package-type-module');
+  });
+
+  it('package.json with "type": "commonjs"', () => {
+    expect('', packageTypeCommonJsMain, 'package-type-commonjs');
+  });
+
+  it('package.json with no "type" field', () => {
+    expect('', packageWithoutTypeMain, 'package-without-type');
+  });
+
+  it('--input-type=module disallowed for files', () => {
+    expect(
+      '--input-type=module',
+      packageTypeModuleMain,
+      'ERR_INPUT_TYPE_NOT_ALLOWED',
+      true,
+    );
+  });
+
+  it('--input-type=module disallowed for directories', () => {
+    try {
+      require('../fixtures/es-modules/package-type-module/index.js');
```

Comment:
shouldn't we `assert.throws` the execution of this statement instead of catching the exception and doing the assert in the catch block?

Reply:
@simoneb yeah didn't think about it 😄 updated!